### PR TITLE
school-library: decorate class

### PR DESCRIPTION
**Pull request by omarbabou**
In this PR, I Implemented composition (as an example of the design pattern) and run it using the command line.
I added the following elements:

  - create a `Nameable` class, Implement a method called `correct_name` that will raise a `NotImplementedError.`
  - Turn the `Person` class to `Nameable` by inheritance.
  - Create a base `Decorator `class that inherits from `Nameable`
  - Create `CapitalizeDecorator` and `TrimmerDecorator `that inherits from `Decorator.`
  - Correct linters errors.

### CODE REVIEW

File: src/person.rb [hunk 1]
```diff
@@ -1,9 +1,12 @@
-class Person
+require_relative './decorate'
+
+class Person < Decorate
```
Comment (TedLivist) [inline src/person.rb]: - [ ] The Person class should inherit from Nameable, not Decorator. Kindly fix that

File: src/decorate.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.split.first(10).join(' ')
+  end
+end
```
Comment (TedLivist) [inline src/decorate.rb]: - [x] **[OPTIONAL]** According to Ruby conventions, classes should be in their files. I strongly recommend you move these 4 classes into 4 different files.

File: src/person.rb [hunk 3]
```diff
@@ -18,4 +21,8 @@ def of_age?
   def can_use_services?
     is_of_age || @parent_permission
   end
```
Comment (TedLivist) [inline src/person.rb]: - [x] I encountered an error when testing the app. You defined `of_age?` (note the '?' also) but you are calling a non-existent `is_of_age`. Also, the method above this one that you assigned private is also `is_of_age?`. Kindly fix that.

  of_age

File: src/decorate.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.split.first(10).join(' ')
```
Comment (TedLivist) [inline src/decorate.rb]: - [x] After fixing the errors, you'd realize your trimmer does not also work as expected. Kindly fix that.

  You can resolve this by refactoring to this:
  ```
  @nameable.correct_name[0...10]
  ```

  tti

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @omarbabou,

Good job so far!
✔️ Good documentation
✔️ Capitalize class works as required
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi @omarbabou , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**